app/services/activitypub/process_account_service.rb

```diff
@@ -99,6 +102,18 @@ def set_fetchable_attributes!
     @account.moved_to_account  = @json['movedTo'].present? ? moved_account : nil
   end
 
+  def set_suspension!
+    return if @account.suspended? && @account.suspension_origin_local?
```

Comment:
Doesn't this mean that in the following scenario, you'll miss the account being suspended?
- you locally suspend a remote account
- the remote account gets suspended on their instance
- you locally unsuspend that account

Reply:
I think this could be solved by doing the following when unsuspending a locally-suspended remote account:
1. change the suspension origin from local to remote (even if the account is not suspended remotely)
2. refresh the account profile (if the account is not suspended remotely, it will be marked as unsuspended)